src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -327,6 +327,16 @@ protected function getLocalUrl($path)
             return trim($config->get('url'), '/').'/'.ltrim($path, '/');
         }
 
+        // If the visibility of the file system is public and the developer chose to use a sub-folder
+        // in the local storage, then we need to grab the sub-folders specified after public for the
+        // symbolic link to fully resolve the file path.
+        if ($config->has('visibility') && $config->get('visibility') == 'public' && ! Str::endsWith($path, 'public')) {
```

Comment:
As `$config->get('visibility')` will return `null` as a default, calling `$config->has` is useless.
This should be just enough:
```php
if ($config->get('visibility') === 'public' && ! Str::endsWith($path, 'public')) {
```